flepimop/gempyor_pkg/src/gempyor/NPI/SinglePeriodModifier.py

```diff
@@ -152,12 +153,12 @@ def __createFromConfig(self, npi_config):
             npi_config, list(self.affected_subpops)
         )
         if self.spatial_groups["ungrouped"]:
-            self.parameters.loc[self.spatial_groups["ungrouped"], "value"] = self.dist(
-                size=len(self.spatial_groups["ungrouped"])
+            self.parameters.loc[self.spatial_groups["ungrouped"], "value"] = (
+                self.dist.sample(size=len(self.spatial_groups["ungrouped"]))
             )
         if self.spatial_groups["grouped"]:
             for group in self.spatial_groups["grouped"]:
-                drawn_value = self.dist(size=1) * np.ones(len(group))
+                drawn_value = self.dist() * np.ones(len(group))
```

Comment:
this is fine as-is, but seems clearer to implement as:

```suggestion
                drawn_value = np.repeat(self.dist(), len(group))
```

might also have some value in maintaining type more clearly?